File: CourtCounter/app/src/main/java/com/example/courtcounter/MainActivity.java
package com.example.courtcounter;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    int scoreForA = 0;
    int scoreForB = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }


    public void displayForTeamA(int score) {
        TextView scoreView = (TextView) findViewById(R.id.team_a_score);
        scoreView.setText(String.valueOf(score));
    }


    public void displayForTeamB(int score) {
        TextView scoreView = (TextView) findViewById(R.id.team_b_score);
        scoreView.setText(String.valueOf(score));
    }


    public void threePointsForA(View view) {
        scoreForA += 3;
        displayForTeamA(scoreForA);
    }


    public void twoPointsForA(View view) {
        scoreForA += 2;
        displayForTeamA(scoreForA);
    }


    public void freeThrowForA(View view) {
        scoreForA += 1;
        displayForTeamA(scoreForA);
    }

    public void threePointsForB(View view) {
        scoreForB += 3;
        displayForTeamB(scoreForB);
    }

    public void freeThrowForB(View view) {
        scoreForB += 1;
        displayForTeamB(scoreForB);
    }

    public void twoPointsForB(View view) {
        scoreForB += 2;
        displayForTeamB(scoreForB);
    }

    public void resetPoints(View view) {
        scoreForA = 0;
        scoreForB = 0;
        displayForTeamA(scoreForA);
        displayForTeamB(scoreForB);
    }
}

File: MoneyView2/app/src/main/java/com/example/moneyview/MainActivity.kt
package com.example.moneyview

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.LayoutInflater
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import androidx.appcompat.app.AlertDialog
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

class MainActivity : AppCompatActivity(), MRVInterface {

    lateinit var viewModel: MoneyViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        viewModel = ViewModelProvider(this ,
                ViewModelProvider.AndroidViewModelFactory.getInstance(application)).get(MoneyViewModel::class.java)

        val recyclerView = findViewById<RecyclerView>(R.id.recycler)
        recyclerView.layoutManager = LinearLayoutManager(this)
        val adapter = MoneyAdapter(this , this , viewModel)
        recyclerView.adapter = adapter

        viewModel.allMoney.observe(this , Observer {list->
            list?.let{
                adapter.update(it)
            }
        })

        val add = findViewById<Button>(R.id.add)
        add.setOnClickListener {
            val etName = findViewById<EditText>(R.id.etName)
            val name = etName.text.toString()
            val etAmount = findViewById<EditText>(R.id.etAmount)
            val amount = etAmount.text.toString().toInt()
            viewModel.insert(Money(name , amount))
        }

    }

    override fun onItemDelete(money: Money) {

        val builder = AlertDialog.Builder(this)
        builder.setTitle("Delete")
                .setMessage("Do you really want to delete.")
                .setPositiveButton("Yes"){dialog, which ->
                    viewModel.delete(money)
                }
                .setNegativeButton("No"){dialog, which ->
                    dialog.cancel()
                }
        builder.show()
    }

}